src/GPU3D.cpp

```diff
@@ -2903,6 +3332,7 @@ void GPU3D::Write32(u32 addr, u32 val) noexcept
     if (addr >= 0x04000400 && addr < 0x04000440)
     {
         WriteToGXFIFO(val);
+        NewWriteFD(addr, val);
```

Comment:
I'm not sure about this, because it would change the order a bit. But it might be slightly faster to only store commands batched in GPU3D::Run()

Reply:
i dont *think* that would change the order? it might also be easier to work with for some more complex features im planning on.
would still require tracking zero dot disp register writes separately though, but i think the order would still work out fine(?)
also would require saving command data to file as 16 bit "id' (256 gx cmds, +1 extra), 32 bit "write" (and probably some manual batching of cmds?)
could get away with still using 8 bits for id if i either redirect one of the nops to another cmd, or repurpose the box/vector tests ids (since i dont think those do anything relevant for this purpose).
though this whole system for logging and storing writes goes out the window if i implement support for 2d engine gfx